baseGeometry tests
==================


	>>> c = __import__("7-base_geometry").BaseGeometry.__doc__
        >>> len(c) > 1
        True
	>>> f1 = __import__("7-base_geometry").BaseGeometry.area.__doc__
        >>> len(f1) > 1
        True
	>>> f2 = __import__("7-base_geometry").BaseGeometry.integer_validator.__doc__
        >>> len(f2) > 1
        True

	>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry
	>>> bg = BaseGeometry()
	>>> bg.integer_validator("my_int", 12)
	>>> bg.integer_validator("width", 89)

	>>> bg.integer_validator("name", "John Cina")
	Traceback (most recent call last):
	   ...
	TypeError: name must be an integer

	>>> bg.integer_validator("fame", 0)
	Traceback (most recent call last):
	   ...
	ValueError: fame must be greater than 0

	>>> bg.integer_validator("f", 2.5)
        Traceback (most recent call last):
           ...
        TypeError: f must be an integer

	>>> bg.integer_validator("f", float('inf'))
        Traceback (most recent call last):
           ...
        TypeError: f must be an integer

	>>> bg.integer_validator("f", None)
        Traceback (most recent call last):
           ...
        TypeError: f must be an integer

	>>> bg.integer_validator("f", [2.5, 6])
        Traceback (most recent call last):
           ...
        TypeError: f must be an integer

	>>> bg.integer_validator("f", True)
        Traceback (most recent call last):
           ...
        TypeError: f must be an integer

	>>> bg.integer_validator("f", (2.5, 6))
        Traceback (most recent call last):
           ...
        TypeError: f must be an integer

	>>> bg.integer_validator("f", {2.5: 6})
        Traceback (most recent call last):
           ...
        TypeError: f must be an integer

	>>> bg.integer_validator("balance", -99)
	Traceback (most recent call last):
	   ...
	ValueError: balance must be greater than 0

	>>> bg.integer_validator("number", -5)
        Traceback (most recent call last):
           ...
        ValueError: number must be greater than 0

	>>> bg.integer_validator()
	Traceback (most recent call last):
	   ...
	TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'
	
	>>> bg.integer_validator('name')
	Traceback (most recent call last):
	   ...
	TypeError: integer_validator() missing 1 required positional argument: 'value'


	>>> bg.area()
	Traceback (most recent call last):
	   ...
	Exception: area() is not implemented
